chromium_src/third_party/blink/renderer/core/html/media/autoplay_policy.cc

```diff
@@ -5,10 +5,14 @@
 
 #include "third_party/blink/public/platform/web_content_settings_client.h"
 
-#define BRAVE_AUTOPLAY_POLICY_IS_GESTURE_NEEDED_FOR_PLAYBACK \
-  if (IsAutoplayAllowedPerSettings())                        \
```

Comment:
Why don't we change this to 
```cpp
if (!IsAutoplayAllowedPerSettings())
   return true;
```
both `RequestAutoplayByAttribute` and `RequestPlay` still check `IsGestureNeededForPlayback`

Reply:
The browser tests (now, correctly) turn off the gesture-needed-for-playback flag, like the upstream media tests, so having our patch inside that function was causing problems. I moved the logic to the callers.

Reply:
I see. @bridiver I think we have to keep the new way of patching because even if we move 
`BRAVE_AUTOPLAY_POLICY_IS_GESTURE_NEEDED_FOR_PLAYBACK` to the beginning of `AutoplayPolicy::IsGestureNeededForPlayback` to reduce the patch, the blocking behavior will be not entirely correct. Look at this code block
```cpp
base::Optional<DOMExceptionCode> AutoplayPolicy::RequestPlay() {
  if (!LocalFrame::HasTransientUserActivation(
          element_->GetDocument().GetFrame())) {
    autoplay_uma_helper_->OnAutoplayInitiated(AutoplaySource::kMethod);
    if (IsGestureNeededForPlayback())
      return DOMExceptionCode::kNotAllowedError;
  } else {
    TryUnlockingUserGesture();
  }
...
}
```
When `LocalFrame::HasTransientUserActivation` is true, it won't check GesturePolicy.
It is easy to see the difference when you block imgur.com and youtube.com